web/tests/calculate-percentage-directive-spec.js

```diff
@@ -42,32 +92,33 @@ describe('calculatePercentage directive', function () {
             this.elm = this.$compile(tpl)(this.scope);
             this.scope.$digest();
         };
-        spyOn(this.calculateCompletionPercent, 'countQuestionsTotal').and.callThrough();
+        spyOn(this.calculateCompletionPercent, 'getImportantQuestionKeys').and.callThrough();
         spyOn(this.calculateCompletionPercent, 'countNumberOfAnsweredQuestions').and.callThrough();
     });
 
-    it('should call this.calculateCompletionPercent.countQuestionsTotal with the questionSet', function () {
+    it('should call this.calculateCompletionPercent.getImportantQuestionKeys with the questionSet', function () {
         this.scope.fields = questionSet;
         this.compileDirective(this.tpl);
-        expect(this.calculateCompletionPercent.countQuestionsTotal).toHaveBeenCalledWith(questionSet)
+        expect(this.calculateCompletionPercent.getImportantQuestionKeys).toHaveBeenCalledWith(questionSet)
     });
 
-    it('should return the total number of questions', function () {
-        const questionCount = this.calculateCompletionPercent.countQuestionsTotal(questionSet);
-        expect(questionCount).toEqual(6);
+    it('should return an array of keys', function () {
+        const keys = this.calculateCompletionPercent.getImportantQuestionKeys(questionSet);
+        expect(keys).toEqual([ 2, 3, 2, 3 ]);
```

Comment:
It is interesting that the keys could be repeated. Maybe it should use a [Set](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Set)? - It would probably simplify the the service code to use `let qKeys = new Set();` ... `qKeys.add(field.key)` and `qKeys.has(answerKey)`